src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -542,6 +510,54 @@ pass::EliminateEltwise::EliminateEltwise() {
     this->register_matcher(m, callback);
 }
 
+NGRAPH_RTTI_DEFINITION(pass::EliminateReshape, "EliminateReshape", 0);
+
+pass::EliminateReshape::EliminateReshape() {
+    MATCHER_SCOPE(EliminateReshape);
+    auto reshape_node_pattern = pattern::wrap_type<opset8::Reshape>();
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher &m) {
+        auto node = m.get_match_root();
+        auto input = node->input_value(0);
+        // check if reshape is not identity op
+        if (input.get_partial_shape().is_dynamic() || node->get_output_partial_shape(0).is_dynamic()) {
+            NGRAPH_DEBUG << node << " has dynamic shapes.";
+            return false;
+        }
+        // remove identity op
+        if (input.get_shape() == node->get_output_shape(0)) {
+            return replace_output_update_name(node->output(0), input);
+        }
+        // eliminate redundant reshape, squeeze, or unsqueeze
+        auto input_node = input.get_node_shared_ptr();
+        if (ov::as_type_ptr<opset3::Squeeze>(input_node) ||
+            ov::as_type_ptr<opset3::Unsqueeze>(input_node) ||
+            ov::as_type_ptr<opset3::Reshape>(input_node)) {
```

Comment:
I added check for consumer count in this transformation and removed Reshape from the condition.
But as I understand, transformation ReshapeSequenceFusion doesn't eliminate redundant reshapes, it always fuses into one reshape. So I updated ReshapeSequenceFusion transformation to cover interchangeable reshapes (to eliminate it)
Thanks!

Reply:
Now I'm not sure about Reshape removing from this condition because transformation `NopElimination` contains `EliminationSqueeze` and `EliminationUnsqueeze` that convert `Squeeze` and `Unsqueeze` to `Reshape`. Moreover `NopElimination` contains `EliminationReshape`. And if we remove Reshape, using `NopElimination` we will have pattern Reshape->Reshape after Squeeze->Reshape, for example. Is it ok for `NopElimination`? What do you think?